src/main/java/com/ace/alfox/controllers/LoginController.java

```diff
@@ -28,24 +27,38 @@ public ResponseEntity getLoginStatus(HttpServletRequest request) {
     }
   }
 
+  /**
+   * The login takes a mapped request body of fields mapped to the strings username and password representing a login attempt.
+   * @param loginRequest
+   * @param request
+   * @return
+   */
   @PostMapping("/login")
-  public ResponseEntity login(@RequestParam String user, HttpServletRequest request) {
+  public ResponseEntity login(@RequestBody Map<String,Object> loginRequest, HttpServletRequest request) {
     HttpSession session = request.getSession(true);
-    Player p = db.players.find(user);
+    Player p = db.players.authenticate(loginRequest.get("username").toString(),loginRequest.get("password").toString());
     if (p == null) {
-      /* this will never happen until Adam rewrites `PlayerRepository.find` to <u>not</u> auto-create test accounts. */
       return ResponseEntity.notFound().build();
     }
-
-    // TODO: Lookup user and verify credentials
-
     session.setAttribute("pid", p.id);
     return ResponseEntity.ok().build();
   }
 
+  /**
+   * Signup takes a similar request to login, checks to see if the username is unique and if it is signs up, then logs the user in.
+   * @param signupRequest
+   * @param request
+   * @return
+   */
   @PostMapping("/signup")
-  public ResponseEntity signup(HttpServletRequest request) {
-    // TODO implement user signup
+  public ResponseEntity signup(@RequestBody Map<String,Object> signupRequest, HttpServletRequest request) {
+    Player p = db.players.find(signupRequest.get("username").toString());
+    if (p==null){
+      p=db.players.signUp(signupRequest.get("username").toString(),signupRequest.get("password").toString());
+      login(signupRequest,request);
```

Comment:
Seems like there is a bit of a code smell here. You're assigning to p (in general full names are always better for maintainability as a side note) but then you ignore p and don't use it. You should either not assign to p, or put in the intended checks (ie for not null or verifying success, etc)